WordPress/src/main/java/org/wordpress/android/ui/sitecreation/usecases/CreateSiteUseCase.kt

```diff
@@ -45,12 +49,18 @@ class CreateSiteUseCase @Inject constructor(
          * time of this comment, changing FluxC's Payload might end up affecting the old site creation flow,
          * so the workaround is applied here instead.
          */
-        val domain = if (isWordPressComSubDomain(siteData.domain)) {
-            urlUtilsWrapper.extractSubDomain(siteData.domain)
-        } else siteData.domain
+        val domain = when {
+            siteData.domain.isNullOrEmpty() -> null
+            isWordPressComSubDomain(siteData.domain) -> urlUtilsWrapper.extractSubDomain(siteData.domain)
+            else -> siteData.domain
+        }
+        val username = accountStore.account?.userName
+        val emailUser = signupUtils.createUsernameFromEmail(accountStore.account.email)
         return suspendCoroutine { cont ->
             val newSitePayload = NewSitePayload(
+                    username ?: emailUser ?: "",
```

Comment:
What is the scenario where both of these would be null? Is this something we can still handle as a payload (as `""`), or an error state of some kind? If the latter, should we log it?

Reply:
I've investigated this further and I don't see any cases where this would be null. The logic is updated with 31301744d733d18fee46f4a2a31fd435140f34e4